DRILL-7830: Add XML as Supported Format for HTTP Plugin

cgivre opened the pull request:
# [DRILL-7830](https://issues.apache.org/jira/browse/DRILL-7830): Add XML as Supported Format for HTTP Plugin

## Description
This PR extends the HTTP/REST plugin to accept XML as input. This PR reuses Drill's existing XML reader.

## Documentation
This PR allows the REST plugin to query REST APIs which return XML data.  The REST plugin currently has an option called `inputType`.  The available options are `xml`, `json`, `csv`.   


This PR also adds an additional configuration option of `xmlDataLevel` which sets the starting nesting level for the XML data.  This helps reduce the amount of unhelpful nesting in the data.  You can view the [documentation for the XML plugin here](https://github.com/apache/drill/tree/master/contrib/format-xml).

## Testing
Added additional unit test.

luocooong commented on contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpApiConfig.java:
```diff
@@ -131,6 +133,8 @@ public HttpApiConfig(@JsonProperty("url") String url,
 
     this.inputType = inputType == null
       ? "json" : inputType.trim().toLowerCase();
```
It is recommended to define "json" as a final static variable.

cgivre replied:
Fixed

luocooong commented on contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpScanBatchCreator.java:
```diff
@@ -99,6 +99,8 @@ public void bind(ManagedScanFramework framework) { }
       if (count++ == 0) {
         if (inputType.equalsIgnoreCase("csv")) {
           return new HttpCSVBatchReader(subScan);
+        } else if (inputType.equalsIgnoreCase("xml")) {
```
Could 'csv' and 'xml' reuse the final static variables recommended above?

cgivre replied:
Fixed

cgivre commented:
Thanks @luocooong for the quick review!  I addressed your changes and squashed commits.   Will wait until tomorrow to merge.
Thanks again!

@dzamo 
There are some minor updates to the docs included in this PR.  I updated the `README` for the HTTP storage plugin to reflect that this can now read XML.

dzamo commented:
Docs updated and will reflect after the next site build.
